securitas/tests/unit/controller/test_authentication.py

```diff
@@ -1,71 +1,171 @@
+from unittest import mock
+
 import pytest
+import python_freeipa
 from bs4 import BeautifulSoup
+from flask import session, get_flashed_messages
 
+from securitas import ipa_admin
 
-def test_logout(client):
-    """Test logout"""
-    # check that when unauthed, we redirect back to /
-    result = client.get('/logout', follow_redirects=True)
-    page = BeautifulSoup(result.data, 'html.parser')
-    assert page.title
-    assert page.title.string == 'Self-Service Portal - The Fedora Project'
+
+@pytest.fixture
+def dummy_user_expired_password():
+    ipa_admin.user_add(
+        'dummy',
+        'Dummy',
+        'User',
+        'Dummy User',
+        user_password='dummy_password',
+        login_shell='/bin/bash',
+    )
+    # Don't change the password: it will be expired
+    yield
+    ipa_admin.user_del('dummy')
+
+
+# Logout
+
+
+def test_logout_unauthed(client):
+    """Test logout when not logged in"""
+    result = client.get('/logout', follow_redirects=False)
+    assert result.status_code == 302
+    assert result.location == "http://localhost/"
+    # Make sure we haven't instructed the user to login in a flash message.
+    messages = get_flashed_messages()
+    assert len(messages) == 0
+
+
+@pytest.mark.vcr()
+def test_logout(client, logged_in_dummy_user):
+    """Test logout with a logged-in user"""
+    result = client.get('/logout', follow_redirects=False)
+    assert result.status_code == 302
+    assert result.location == "http://localhost/"
+    assert "securitas_session" not in session
+    assert "securitas_username" not in session
+    assert "securitas_ipa_server_hostname" not in session
+
+
+# Login
 
 
 @pytest.mark.vcr()
 def test_login(client):
     """Test a successful Login"""
     result = client.post(
         '/login',
-        data={"username": "dudemcpants", "password": "n:nPv{P].9}]!q$RE%w<38@",},
+        data={"username": "dudemcpants", "password": "n:nPv{P].9}]!q$RE%w<38@"},
```

Comment:
shouldn't this be the dummy user now? the cassettes might be hiding this failing.

Reply:
Correct, it's only working because we didn't update the cassettes. Also, it should use the `dummy_user` fixture.